optaplanner-core/src/test/java/org/optaplanner/core/impl/localsearch/decider/acceptor/greatdeluge/GreatDelugeAcceptorTest.java

```diff
@@ -0,0 +1,275 @@
+        moveScope3.setScore(HardMediumSoftScore.of(-5,-50,-100));
+        LocalSearchMoveScope<TestdataSolution> moveScope4 = new LocalSearchMoveScope<>(stepScope0, 0, mock(Move.class));
+        moveScope4.setScore(HardMediumSoftScore.of(0,-22,-200));
+
+        assertEquals(true, acceptor.isAccepted(moveScope0));
+        assertEquals(false, acceptor.isAccepted(moveScope1));
+        assertEquals(true, acceptor.isAccepted(moveScope2));
+        assertEquals(false, acceptor.isAccepted(moveScope3));
+        assertEquals(true, acceptor.isAccepted(moveScope4));
+
+        stepScope0.setStep(moveScope4.getMove());
+        stepScope0.setScore(moveScope4.getScore());
+        solverScope.setBestScore(moveScope4.getScore());
+        acceptor.stepEnded(stepScope0);
+        phaseScope.setLastCompletedStepScope(stepScope0);
+
+        acceptor.phaseEnded(phaseScope);
+    }
+
+    @Test
+    public void isAcceptedPositiveLevelMultipleScoreRainSpeedRatio() {
+
+        GreatDelugeAcceptor acceptor = new GreatDelugeAcceptor();
+        acceptor.setInitialLevels(HardMediumSoftScore.of(0, 200, 500));
+        acceptor.setRainSpeedRatio(0.9);
+
+        DefaultSolverScope<TestdataSolution> solverScope = new DefaultSolverScope<>();
+        solverScope.setBestScore(HardMediumSoftScore.of(0, -200, -1000));
+        LocalSearchPhaseScope<TestdataSolution> phaseScope = new LocalSearchPhaseScope<>(solverScope);
+        LocalSearchStepScope<TestdataSolution> lastCompletedStepScope = new LocalSearchStepScope<>(phaseScope, -1);
+        lastCompletedStepScope.setScore(HardMediumSoftScore.of(0, -200, -1000));
+        phaseScope.setLastCompletedStepScope(lastCompletedStepScope);
+        acceptor.phaseStarted(phaseScope);
+
+
+        // lastCompletedStepScore = 0/-200/-1000
+        // water level 0/180/450
+        LocalSearchStepScope<TestdataSolution> stepScope0 = new LocalSearchStepScope<>(phaseScope);
+        acceptor.stepStarted(stepScope0);
+        LocalSearchMoveScope<TestdataSolution> moveScope0 = new LocalSearchMoveScope<>(stepScope0, 0, mock(Move.class));
+        moveScope0.setScore(HardMediumSoftScore.of(0,-180,-300));
+        LocalSearchMoveScope<TestdataSolution> moveScope1 = new LocalSearchMoveScope<>(stepScope0, 0, mock(Move.class));
+        moveScope1.setScore(HardMediumSoftScore.of(0,-180,-500));
+        LocalSearchMoveScope<TestdataSolution> moveScope2 = new LocalSearchMoveScope<>(stepScope0, 0, mock(Move.class));
+        moveScope2.setScore(HardMediumSoftScore.of(0,-50,-800));
+        LocalSearchMoveScope<TestdataSolution> moveScope3 = new LocalSearchMoveScope<>(stepScope0, 0, mock(Move.class));
+        moveScope3.setScore(HardMediumSoftScore.of(-5,-50,-100));
+        LocalSearchMoveScope<TestdataSolution> moveScope4 = new LocalSearchMoveScope<>(stepScope0, 0, mock(Move.class));
+        moveScope4.setScore(HardMediumSoftScore.of(0,-180,-450));
+
+        assertEquals(true, acceptor.isAccepted(moveScope0));
```

Comment:
There is: http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertTrue(boolean)

Also seeing that multiple move scopes are asserted below, soft assertions could do a better job, so that these can be verified in a bulk.

Reply:
Can we do AssertJ in OptaPlanner? I thought we can not. Or is there some way to do that in pure JUnit?

Reply:
I did it this way, because of SA acceptorTest, where it is used like this: `assertEquals(true, acceptor.isAccepted(moveScope1));`
Also, I used it for all tests. May change it.

Reply:
optaplanner-core already depends on AssertJ and there are some tests using it. 

@ge0ffrey 's intention was to migrate to junit5 first before we migrate to AssertJ, so that there is only a single source of assertions in OptaPlanner.

However, this test is a perfect fit for soft assertions and every tutorial recommends using AssertJ for that. In opinion it's reasonable to use it here.